jobs/resource_job.go

```diff
@@ -321,6 +337,9 @@ func (a JobsAPI) Restart(id string, timeout time.Duration) error {
 func (a JobsAPI) Create(jobSettings JobSettings) (Job, error) {
 	var job Job
 	jobSettings.sortTasksByKey()
+	if jobSettings.GitSource != nil && jobSettings.GitSource.Provider == "" {
+		jobSettings.GitSource.Provider = workspace.GetProviderFromUrl(jobSettings.GitSource.Url)
```

Comment:
rename `GetProviderFromUrl` to `GetGitProviderFromUrl` to have clearer semantics of exported functions